source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -645,7 +645,12 @@ void GlslShaderGenerator::emitVariableDeclaration(const ShaderPort* variable, co
     }
     else
     {
-        string str = qualifier.empty() ? EMPTY_STRING : qualifier + " ";
+        string str = "";
+        if (context.getOptions().textureSpaceRender && stage.getName() == "vertex" && qualifier == "in")
```

Comment:
I agree with Bernard. And supporting layout location for uniforms seems useful in general.

One option would be to let the location index be a signed integer, with -1 default, and then emit layout location if the index is positive. Then the texture baking specific handling here could be moved out from this general method.

Another option would be to _always_ emit layout locations for GLSL, but I'm not sure if that has implications elsewhere in other shader generators.

A third option would be to sub-class GlslShaderGenerator and add a derived shader generator specific for texture baking. This might also improve some of the other places where texture baking has been added in, to avoid feature creep into the general code. But again I done fully understand why the shader needs to be different for texture rendering compared to frame rendering?

Reply:
Removed.